### PR TITLE
Wrap on fail clause's content within a statement expression

### LINKED ISSUE #31246: NPE at BIR when continue/break within lambda wrap
**Description:**
```ballerina
import ballerina/io;
public function main() {
    int[] arr = [1, 2, 3];
    foreach var item in arr {
        io:print(item);
        do {
            fail error("");
        } on fail error varName {
            continue;
        }
    }
}

```

**Steps to reproduce:**

**Affected Versions:**
SL beta 1

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by pcnfernando**
This lead us to restrict the usage of statements that changed the flow of control across blocks such as break and continue within onfail/transaction.

Fixes #31246

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-unit-test/src/test/resources/test-src/statements/onfail/on-fail-clause.bal [hunk 1]
```diff
@@ -262,6 +267,74 @@ function testBreakWithinOnfail() {
...
+function testBreakWithinOnfailForOuterLoop() {
+    string str = "";
+    foreach int digit in 1...5 {
+        do {
+            if (digit > 3) {
+                fail getError();
+            }
+        } on fail error e {
+            str += "Loop broke with digit: " + digit.toString();
+            break;
```
Comment (gimantha) [inline tests/jballerina-unit-test/src/test/resources/test-src/statements/onfail/on-fail-clause.bal:313]: Shall we have a test case for continue as well?

Comment (pcnfernando) [reply tests/jballerina-unit-test/src/test/resources/test-src/statements/onfail/on-fail-clause.bal]: Refactored to test both continue and brake flows in b43693abd7969d50dcc6cfecb07ccffce07a870a

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java [hunk 2]
```diff
@@ -4994,15 +4982,43 @@ private void analyzeOnFailClause(BLangOnFailClause onFailClause, BLangBlockStmt
...
+        BVarSymbol onFailErrorVariableSymbol =
+                ((BLangSimpleVariableDef) onFailClause.variableDefinitionNode).var.symbol;
+        BLangSimpleVariable errorVar = ASTBuilderUtil.createVariable(onFailErrorVariableSymbol.pos,
+                onFailErrorVariableSymbol.name.value, onFailErrorVariableSymbol.type, rewrite(fail.expr, env),
+                onFailErrorVariableSymbol);
+        BLangSimpleVariableDef errorVarDef = ASTBuilderUtil.createVariableDef(onFailClause.pos, errorVar);
+        env.scope.define(onFailErrorVariableSymbol.name, onFailErrorVariableSymbol);
+        onFailBody.stmts.add(0, errorVarDef);
+
+        if (onFailClause.isInternal && fail.exprStmt != null) {
```
Comment (pcnfernando) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java:5003]: This flow comes under transaction statements. Hence tests are already in the transaction internal repo

**Review by gimantha — APPROVED** [review]

**Review by warunalakshitha — APPROVED** [review]
BirGen change LGTM

**Pull request merged**